app/controllers/api/v1/recipe_controller.rb

```diff
@@ -10,8 +10,14 @@ def index
       @recipes = Recipe.search_by_keyword(params[:search])
     else
       @recipes = Recipe.all
+    
+      recipes_with_user = @recipes.map do |recipe|
+        recipe_attributes = recipe.attributes
+        recipe_attributes[:user_name] = "#{recipe.user.first_name}  #{recipe.user.last_name}"
```

Comment:
So because of the way ActiveRecord (the ORM that Rails uses to access the database) works, this loop is going to cause the n+1 query problem, where we'll perform 1 query to get a list of records, then an additional query for each recipe to fetch the user data. If there are 2 recipes in the DB, we'll do 3 queries. If there are a thousand recipes, we'll do 1001 queries.

You might check out https://guides.rubyonrails.org/active_record_querying.html#eager-loading-associations for a solution.

The other piece is that we should just return the whole user object, and let the front end figure out what to display. Right now, this makes the backend code tightly coupled to what we want to display on the front end, which can become fragile and hard to maintain over time.

Reply:
We should probably include the user when we search, too, to keep the data format produced by this endpoint the same.